packages/components/tests/dummy/app/templates/components/code-block.hbs

```diff
@@ -0,0 +1,311 @@
+  }
+}"
+  />
+  {{! template-lint-enable no-whitespace-for-layout }}
+
+  <Shw::Text::h3>with hasCopyButton=true</Shw::Text::h3>
+
+  <Hds::CodeBlock
+    id="clipboardTarget"
+    @hasCopyButton={{true}}
+    @language="go"
+    @value="package main
+import 'fmt'
+func main() {
+  fmt.Println('hello world')
+}"
+  />
+
+  <Shw::Text::h3>isReadOnly=false (so content is editable)</Shw::Text::h3>
+
+  <Hds::CodeBlock
+    @isReadOnly={{false}}
+    @value="let codeLang='JavaScript';
+console.log(`I am ${codeLang} code`);"
+  />
+
+  <Shw::Text::h2>Highlight lines</Shw::Text::h2>
+  <Shw::Text::Body>Specify line numbers or ranges of line numbers to highlight</Shw::Text::Body>
+
+  <Shw::Flex @columns={{2}} {{style gap="2rem"}} as |SG|>
+    <SG.Item {{style width="calc(50% - 1rem)"}} @label="Highlight lines 2 & 4">
+      <Hds::CodeBlock
+        @highlightLines={{"2, 4"}}
+        @value="import Application from '@ember/application';
+import Resolver from 'ember-resolver';
+import loadInitializers from 'ember-load-initializers';
+import config from 'dummy/config/environment';
+
+export default class App extends Application {
+  modulePrefix = config.modulePrefix;
+  podModulePrefix = config.podModulePrefix;
+  Resolver = Resolver;
+}
+
+loadInitializers(App, config.modulePrefix);"
+      />
+    </SG.Item>
+
+    <SG.Item {{style width="calc(50% - 1rem)"}} @label="Highlight lines 6-10">
+      <Hds::CodeBlock
+        @highlightLines={{"6-10"}}
+        @value="import Application from '@ember/application';
+import Resolver from 'ember-resolver';
+import loadInitializers from 'ember-load-initializers';
+import config from 'dummy/config/environment';
+
+export default class App extends Application {
+  modulePrefix = config.modulePrefix;
+  podModulePrefix = config.podModulePrefix;
+  Resolver = Resolver;
+}
+
+loadInitializers(App, config.modulePrefix);"
+      />
+    </SG.Item>
+  </Shw::Flex>
+
+  <Shw::Text::h3>language</Shw::Text::h3>
+
+  <Shw::Text::Body>
+    Specify coding language syntax to use<br />
+    <mark>
+      TODO: Figure out how to get AutoLoader plugin working to dynamically load language modules
+    </mark>
+  </Shw::Text::Body>
+
+  <Shw::Flex @columns={{2}} {{style gap="2rem"}} as |SG|>
+    <SG.Item {{style width="calc(50% - 1rem)"}} @label="JavaScript">
+      <Hds::CodeBlock
+        @language="javascript"
+        @value="let codeLang='JavaScript';
+console.log(`I am ${codeLang} code`);"
+      />
+    </SG.Item>
+
+    <SG.Item {{style width="calc(50% - 1rem)"}} @label="Go">
+      <Hds::CodeBlock
+        @language="go"
+        @value="package main
+import 'fmt'
+func main() {
+  fmt.Println('hello world')
+}"
+      />
+    </SG.Item>
+
+    <SG.Item {{style width="calc(50% - 1rem)"}} @label="Markdown">
+      {{! template-lint-disable no-whitespace-for-layout }}
+      <Hds::CodeBlock
+        @language="markdown"
```

Comment:
I know this markdown is pulled from our docs site, but it might be better to be something more representative of actual markdown that a consumer is going to use. Here's an example that I've been testing with that contains more or less all different types of tokens/content:

```
# Here is some inline code

<p>Here is some text within a paragraph tag</p>

## Long markdown content

Lorem ipsum dolor sit amet, consectetur adipiscing elit, 
sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
culpa qui officia deserunt mollit anim id est laborum.

### Level 3 headline

- Item 1
- Item 2
- Item 3

1. Number one
2. Number two
3. Number three

## Here is a table

| Element     | Usage       |
| ----------- | ----------- |
| Icon        | Required    |
| Description | Required    |

Lorem ipsum dolor sit amet, consectetur [and a link](https://hashicorp.com) adipiscing elit

![Alternative text for an image](/public/assets/image.png)
```

Reply:
Thanks, changing. Let me know if you have suggestions to improve any of the other examples also btw.